### PR TITLE
Add UI for a setting to turn off double click to edit project file

### LINKED ISSUE #4493: We need a setting to define the what double clicking project does
Version: 16.0 Preview 2

### Steps to reproduce:

1. Open a solution with a CPS-based C# project
2. Double click to expand the project in Solution Explorer.

### Expected results:

The project expands in Solution Explorer.

### Actual results:

The project does not expand in Solution Explorer, and instead the project opens.

📝 I was aware of the behavior change but avoided filing this issue until I actually hit it by accident. It happened within 5 minutes of installing Preview 2.

### PR DESCRIPTION
**Pull request by davidwengier**
The code that checks this setting is in CPS.

Fixes #4493.

Note: This may not go ahead as the UI may be moved to the VS repo.

### CODE REVIEW

File: src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPageResources.resx [hunk 1]
```diff
@@ -138,4 +138,7 @@
   <data name="General_UsePreviewSdk" xml:space="preserve">
     <value>Use previews of the .NET Core SDK</value>
   </data>
+  <data name="General_DoubleClickOpenProjectFile" xml:space="preserve">
+    <value>Double click on project node to edit project file.</value>
```
Comment (davkean) [inline src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPageResources.resx]: This should not have a trailing ".".

Comment (davkean) [reply src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPageResources.resx]: Also missing an access key.

File: src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPageControl.xaml [hunk 2]
```diff
@@ -17,6 +17,9 @@
             <StackPanel Orientation="Horizontal">
                 <CheckBox x:Name="UsePreviewSdk" x:Uid="General_UsePreviewSdk" Content="{x:Static r:GeneralOptionPageResources.General_UsePreviewSdk}" Margin="13, 10" />
             </StackPanel>
+            <StackPanel Orientation="Horizontal">
+                <CheckBox x:Name="DoubleClickOpenProjectFile" x:Uid="General_DoubleClickOpenProjectFile" Content="{x:Static r:GeneralOptionPageResources.General_DoubleClickOpenProjectFile}" Margin="13, 10" />
+            </StackPanel>
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPageControl.xaml]: Stack panels with a single child aren't necessary.

File: src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPageResources.resx (hunk 1 above)
Comment (davidwengier) [reply src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPageResources.resx]: My WPF is weak... pointers?
image
image

Comment (drewnoakes) [reply src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPageResources.resx]: Use a single underscore instead of an ampersand.

Comment (davidwengier) [reply src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPageResources.resx]: WPF is weird. Bring back Windows Forms I say!

Thanks guys :P

Comment (drewnoakes) [reply src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPageResources.resx]: `&` has a special meaning in XML. You'd have to use `&amp;`.

**Comment by davkean** [comment]
Given https://dev.azure.com/devdiv/DevDiv/_workitems/edit/779596?src=WorkItemMention&src-action=artifact_link, and the more I think about this, I'm thinking we might want to push this setting along with the Preview settings instead of shipping the org chart here.

**Comment by davidwengier** [comment]
Hadn't seen that. Makes the UI tricky, as a setting for previewing .NET SDK project files would apply to all three of the current preview window sub-settings, and I would have thought we'd still want a separate setting for the double click behaviour.

image

**Comment by davkean** [comment]
I'd say that it's only relevant for Solution Explorer, though I'd need to ask the customer what exactly bothers them about it.

If you wanted to switch both, it's weird that one would be under .NET Core and one under that.

**Comment by davidwengier** [comment]
For sure the Tabs and Windows section is a better spot for both settings, just not sure what layout would make sense.

**Comment by davidwengier** [comment]
Renamed the setting based on feedback from the UX meeting today, and changed a couple of other strings even though they're not used, because they might be once #4561 is done.

**Comment by davidwengier** [comment]
UX board has recommended against adding an option to control this behaviour.

**Pull request closed without merging**